### DISCUSSION TITLE
Can reactomeGSA r package accept list of gene names with logfold values?

### DISCUSSION THREAD

**Discussion by KikoSchmiko** [reactome/ReactomeGSA#83] [Q&A] [answered]
Hello everyone!

Thank you in advance for your time and help!

Is there a way to submit a list of genes (with or without fold change values) and to have a list of pathways the genes are enriched in returned to you in r? Something similar to what you get via the analysis feature of the reactome website, but that can be done in R instead? I have already viewed the vignette but am confused about needing an Elist/ am wondering how to provide a dataframe of genes and fold change values.

I hope this makes sense!
Thank you so much for your help!

**Comment by alexgrent** [comment, marked as answer]
Hello @KikoSchmiko,

we don't provide an R Client for the analysis. However, you can access the analysis feature also via our public [API documentation](https://reactome.org/AnalysisService/#/identifiers/getPostText_1) and integrate it within your R workspace.

Further, to perform pathway analysis you only need the raw or normalized counts and the workflow provided by the [Main vignette](https://bioconductor.org/packages/release/bioc/vignettes/ReactomeGSA/inst/doc/using-reactomegsa.html)

Kind regards,
Alexander

**Reply by KikoSchmiko** [reply]
Thank you so much! Your answer helped a lot and I was able to use httr to communicate with the API to run my analysis! 

I had a follow up question; I have pathway results back, is there any API way to map these pathways to their broader associated function such as when there is the hierarchical clustering on the website? I looked through the API support page and didn't see anything but want to make sure I'm not missing it!

Thank you so much for your help!!!

**Reply by EliotRagueneau** [reply]
Dear @KikoSchmiko 

You can access the ancestry of any events (reactions or pathways) using the following endpoint: https://reactome.org/ContentService/data/event/{id}/ancestors 
as described in our ContentService docmentation:
https://reactome.org/ContentService/#/events/getEventAncestors 

Hope this will help you 😄
